mrmustard/math/lattice/neighbors.py

```diff
@@ -29,7 +29,8 @@
 def all_neighbors(pivot: tuple[int, ...]) -> Iterator[tuple[int, tuple[int, ...]]]:
     r"""yields the indices of all the neighbours of the given index."""
     for j in range(len(pivot)):  # pylint: disable=consider-using-enumerate
-        yield j, tuple_setitem(pivot, j, pivot[j] - 1)
+        if pivot[j] != 0:
```

Comment:
Were the results actually wrong, or are they wrong only when you remove the k dependence? Because when k-1 goes negative usually the value is multiplied by sqrt(k), which should be zero.

Reply:
Sorry for the double communication, but it should indeed be ok for the vanilla strategy using the current form of the recurrence relation. For the renormalized recurrence relation, however, the sqrt(k) drops out.
Wouldn't it be better to check the values of the pivot index before lowering them by 1? Something like this might pop up with future Fock strategies.